### PR TITLE
[admin][txn] Add transaction admin to get recover time in stats.

### PR DESCRIPTION
**Pull request by liangyepianzhou**
### Motivation
When we use the transaction, we need to determine whether the components of the transaction are ready. During operation and maintenance, it is also necessary to monitor the recovery time of each component of the transaction to determine its impact on performance.
### Modification
Optimize admin tools to get recovery timestamps in TB, TP, TC stats.

### About test
Because it is difficult to interfere with recovery, no validation for `recoverEndTime=-1` was added。

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `no-need-doc` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-added`
(Docs have been already added)

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java [hunk 1]
```diff
@@ -98,6 +99,9 @@ public class TopicTransactionBuffer extends TopicTransactionBufferState implemen
 
     private final ConcurrentHashMap<Long, Long> lowWaterMarks = new ConcurrentHashMap<>();
 
+    //Store transaction buffer recovery start time and end time. The left is start time and the right is end time.
+    public final MutablePair<Long, Long> recoverTimePair = new MutablePair<>(0L, 0L);
```
Comment (congbobo184) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java:103]: can create a Class RecoverTimeRecord to record this

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/TransactionBufferStats.java [hunk 2]
```diff
@@ -28,4 +28,9 @@ public class TransactionBufferStats {
 
     /** The last snapshot timestamps of this transaction buffer. */
     public long lastSnapshotTimestamps;
+
+    //Start timestamp of  transaction buffer recovery. 0L means no startup.
+    public long recoverStartTime;
+    //End timestamp of transaction buffer recovery. 0L means no startup, -1L means exceptionally.
```
Comment (congbobo184) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/TransactionBufferStats.java:34]: don't need set -1, TxnBufferState has already expressed, same above pendingAck and Tc

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/PendingAckHandleImpl.java [hunk 3]
```diff
@@ -147,6 +150,7 @@ private void initPendingAckStore() {
                 this.pendingAckStoreFuture =
                         pendingAckStoreProvider.newPendingAckStore(persistentSubscription);
                 this.pendingAckStoreFuture.thenAccept(pendingAckStore -> {
+                    recoverTime.setRecoverStartTime(System.currentTimeMillis());
```
[the remark this answers is not in the document]
Comment (liangyepianzhou) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/PendingAckHandleImpl.java:153]: Transaction pending ack is lazy loading. There may not start to recover.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/PendingAckHandleImpl.java [hunk 4]
```diff
@@ -861,18 +865,26 @@ public CompletableFuture<PendingAckHandle> pendingAckHandleFuture() {
...
+        transactionPendingAckStats.recoverStartTime = recoverTime.getRecoverStartTime();
+        transactionPendingAckStats.recoverEndTime = recoverTime.getRecoverEndTime();
         return transactionPendingAckStats;
     }
 
     public synchronized void completeHandleFuture() {
         if (!this.pendingAckHandleCompletableFuture.isDone()) {
             this.pendingAckHandleCompletableFuture.complete(PendingAckHandleImpl.this);
         }
+        if (recoverTime.getRecoverStartTime() == 0L) {
```
[the remark this answers is not in the document]
Comment (liangyepianzhou) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/PendingAckHandleImpl.java:877]: Transaction pending ack is lazy loading. There may not start to recover.

**Review by congbobo184 — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**